src/robotlibcore.py

```diff
@@ -142,6 +135,38 @@ def get_keyword_documentation(self, name):
             doc = '{}\n\n{}'.format(doc, tags) if doc else tags
         return doc
 
+    def get_keyword_types(self, keyword_name):
+        method = self.keywords.get(keyword_name)
+        if keyword_name == '__init__':
+            method = self.__init__
+        elif keyword_name.startswith('__') and keyword_name.endswith('__'):
+            return {}
+        if not method:
+            raise ValueError('%s is not keyword.' % keyword_name)
+        robot_types = getattr(method, 'robot_types', ())
+        if robot_types is None:
+            return robot_types
+        annotations = self._get_annotations(method)
+        types = self._join_defaults_with_annotations(method, annotations)
+        if robot_types:
+            types.update(robot_types)
```

Comment:
The logic above doesn't look 100% right to me. It should be like this:

```python
types = getattr(method, 'robot_types', ())
if types is None:
    return types
if not types:
    types = self.__get_annotations(method)
return self.__join_types_with_defaults(types, method)
```

You can see how this is handled on Robot side [here](https://github.com/robotframework/robotframework/blob/master/src/robot/running/arguments/argumentparser.py#L89). Notice that Robot handles defaults differently.